server/lib/accounts.js

```diff
@@ -216,6 +217,21 @@ Accounts.insertUserDoc = _.wrap(Accounts.insertUserDoc, function(insertUserDoc,
 
 	addUserRoles(_id, roles);
 
+	if (settings.get('Accounts_SetDefaultAvatar') === true) {
```

Comment:
This is the only portion I don’t like... but it’s needed to add on creation where the username is set by the so provider.  If the username is not set then the logic in the setUsername flow works and sets the avatar